### PR TITLE
[FIX] point_of_sale: fix exchange-type pos session

### PR DESCRIPTION
**Pull request by agr-odoo**
- Enable Automated Inventory Valuation on a product category (i.e.
Office Furniture)
- Open POS session
- Add -1 of Office Chair Black, 1 Office Chair, as to simulate an
exchange of product
- Close and Validate session

In the Journal Items the entry related to the return is missing. Even if
in stock the return is accouted correctly the journal entry related to
expenses is not updated correclty.
This occur because POS does not take into account the 2 separate
pickings when creating the account move at validation phase, but only
look for the 'positive' picking.

opw-2267977

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by agr-odoo** [comment]
@pimodoo 
I guess this has been fixed for saas-13.1+ with 2f2bde5d9f647e4785ffabdc1b2e07a08722309c
May I have your opinion on this?
The proposed **draft** should display the right numbers, but actually a full reconcile is missing somewhere ( in the second screenshot there's a missing entry for the black chair refund, 18$ reconciled). 
cc @nim-odoo 

**13 without fix**
13-no-fix
**13 with fix**
13-fix
**saas-13.3 already fixed**

**Comment by nim-odoo** [comment]
Tests are failing

**Comment by agr-odoo** [comment]
exactly, this pr is still a draft intended to showcase the issue to someone more knowledgeable on the actual implementation, otherwise accounting will be messed up

**Comment by agr-odoo** [comment]
Edit: now the accounting is working but there may be corner case and I guess the performance of those search + mapped + filtered are not great
**13.0 fix v2**
13-fix2

**Comment by nim-odoo** [comment]
@caburj what do you think about this approach?

**Comment by caburj** [comment]
@agr-odoo @nim-odoo I agree with this solution. There should have been a return picking field from the start as a return picking is created for negative order lines.

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -416,11 +416,7 @@ def _accumulate_amounts(self, data):
 
                 if self.company_id.anglo_saxon_accounting and order.picking_id.id:
                     # Combine stock lines
-                    stock_moves = self.env['stock.move'].search([
-                        ('picking_id', '=', order.picking_id.id),
-                        ('company_id.anglo_saxon_accounting', '=', True),
-                        ('product_id.categ_id.property_valuation', '=', 'real_time')
-                    ])
+                    stock_moves = self.env['stock.picking'].search([('origin', '=', order.name)]).mapped('move_lines').filtered(lambda move: move.company_id.anglo_saxon_accounting and move.product_id.categ_id.property_valuation == "real_time")
```
Comment (caburj) [inline addons/point_of_sale/models/pos_session.py:419]: Hi @agr-odoo. If performance is a concern, I guess we should keep the old code, except we change the `('picking_id', '=', order.picking_id.id)` to `('picking_id', 'in', self.env['stock.picking'].search([('origin', '=', order.name)]).ids)`. What do you think?

**Comment by agr-odoo** [comment]
@caburj suggestion accepted, I refactored to let it be more clear and similar to the original

**Comment by nim-odoo** [comment]
fw-bot ignore

**Pull request closed without merging**